elastic/elastic-agent
# Fix drop processor for monitoring events

## What does this PR do?

It fixes the drop processor for monitoring component logs, instead of using the dataset that does not include any information about whether the component is a monitoring component it now uses the `component.id`.

## Why is it important?

There can be an infinity loop of a monitoring component failing to send an event, logging it, then having it picked up from the logs which will be sent and fail, creating an infinity loop of error messages.

## Checklist


~~## Author's Checklist~~
## How to test this PR locally

Run Elastic-Agent with monitoring enabled, then go to Kibana and look at the dataview `logs-*`, there should be no logs where `component.id` matches the regexp `.*-monitoring$`.

## Related issues

- Closes #2388 

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
